src/main/java-templates/org/postgresql/Driver.java

```diff
@@ -411,7 +411,7 @@ public Connection getResult(long timeout) throws SQLException {
      * @throws SQLException if the connection could not be made
      */
     private static Connection makeConnection(String url, Properties props) throws SQLException {
-        return new @JDBCCONNECTCLASS@(hostSpecs(props),
+        return new ${JDBCCONNECTCLASS}(hostSpecs(props),
```

Comment:
+1 for discussing this is separate threads.

Java 8 and Fixes: Shouldn't be a problem if a fix is developed in a branch and that branch is then merged or cherry-picked into the others. We do that a lot and in 90% of all cases it works well thanks to git. The scenario where a fix is needed in a place that works differently in Java 6 vs Java 8 means duplicate work, right. But how often does that really happen? And does the small adaption really outweigh the stress we have with the current pre-processing solution?

Single Driver: I know SQL Anywhere pretty well and there the driver name changed more often than just with each JDBC level... They even changed the package names frequently. Nobody complained about that. In fact, I even would love to have an explicit name likd "org.postgresql.jdbc3.Driver" vs "org.postgresql.jdbc4.Driver" as then it is really pretty obvious what features I really can expect to get.